src/main/java/newbank/server/Account.java

```diff
@@ -1,11 +1,24 @@
 package newbank.server;
 
-public class Account {
+import newbank.server.exceptions.AccountInvalidNameException;
 
+public class Account {
   private String accountName;
   private double openingBalance;
 
-  public Account(String accountName, double openingBalance) {
+  public static final int MIN_NAME_LENGTH = 4;
+  public static final int MAX_NAME_LENGTH = 12;
+
+  public Account(String accountName, double openingBalance) throws AccountInvalidNameException {
+    if (accountName.length() < MIN_NAME_LENGTH || accountName.length() > MAX_NAME_LENGTH) {
```

Comment:
Those "if" statements belong at a different level of abstraction. I would refactor these in a "validateAccount()" method or something that does the validation.

Kindest Regards,
Jules

Reply:
I see your point, Jules. But don't you think it's dangerous to allow creating an account with an invalid name? Someone could use it without the knowledge it's incorrect.

Reply:
Absolutely. What I meant was something like
public Account(...) {
   validateAccount();
   // carry on as before
}

So those "if" statements would be hidden in the validateAccount() method.

Kindest Regards,
Jules

Reply:
Done! :)